files/routeless-files/index.js

```diff
@@ -1,4 +1,5 @@
-/* eslint-env node */
+/* eslint-disable ember/avoid-leaking-state-in-ember-objects */
```

Comment:
I'd like to resolve these before moving forward

Reply:
Yup! could you maybe provide some guidance on how this can be resolved? [I made a suggestion here](https://github.com/ember-cli/eslint-plugin-ember/issues/251#issuecomment-798109900), but I'm unsure if that's a solid approach.

Reply:
Firstly, that issue is different from what you are describing here. That issue is about `addon/index.js` in an Engine and this comment is about the top level `index.js` (e.g. the "addon entry point). 

---

I think the fix is to avoid the appearance of this being an Ember.Object. There is absolutely no reason that we use `EngineAddon.extend()`, it is just a plain function we are invoking.

I think we can alias the export, and allow:

```js
const { buildEngine } = require('ember-engines/lib/engine-addon');

module.exports = buildEngine({
  name: require('./package.json').name,

  lazyLoading: {
    enabled: false,
  },
});
```

Reply:
Yes, you're right. I was confusing the one with the other. Would it be okay if we create a ticket for your suggestion and I revert the `eslint-disable` comments for now? I can pick up that ticket some other time.

Reply:
ya

Reply:
@rwjblue Created a PR for that new function: https://github.com/ember-engines/ember-engines/pull/758/files?diff=split&w=1

Reply:
Hi @rwjblue! Updated this PR to make use of the new `buildEngine` function.